### DISCUSSION TITLE
Provide a reason of failed auto merge

### DISCUSSION THREAD

**Discussion by havenchyk** [community/community#10621] [Pull Requests]
I enabled auto-merge on my PR and got "An error occurred while performing auto-merge". This message is cryptic and doesn't allow me to get the reason, what was wrong with it. 

It would be nice to see the exact reason somehow, like it's done for GitHub Actions for example

**Comment by FlorianHockmann** [comment]
Would be great to at least get some possible reasons which could lead to this error.

@havenchyk: Did you find out why it didn't work in your case or did you maybe find some workaround?

edit: In our case the problem was that we had the option enabled: `Require linear history` which probably didn't work with auto merge as that creates a merge commit. This problem is also described [here](https://github.com/renovatebot/renovate/issues/12363).

**Comment by kubarozycki** [comment]
Hello, 
Same issue here, how to troubleshoot?

- auto merge enabled in settings
- require linear history disabled

**Comment by dhruv8433** [comment]
@havenchyk 

When you encounter an error message like "An error occurred while performing auto-merge" on GitHub, it can indeed be frustrating due to its lack of specificity. However, there are a few common reasons why an auto-merge might fail, and there are steps you can take to diagnose and resolve the issue.

## Common Reasons for Auto-Merge Failures
1. Merge Conflicts: This is the most common reason. If the changes in your PR conflict with the base branch, auto-merge will fail.
2. Failing Checks: If your PR includes status checks (e.g., continuous integration tests, linters) and any of these checks fail, auto-merge will not proceed.
3. Required Reviews: If your repository is configured to require reviews and those reviews are not yet completed, auto-merge will not occur.
4. Permissions Issues: If there are issues with your permissions or the permissions of the bot/account performing the auto-merge, it may fail.
5. Branch Protection Rules: If there are branch protection rules (e.g., requiring the latest version of the base branch to be merged before the PR can be merged), these might prevent auto-merge.

**Comment by bringhurst** [comment]
I just ran into this "Failed enabling auto-merge for pull request" error because my commit message was too long. Once I reduced the size of the commit message, enabling auto-merge worked.

**Reply by shaunburch** [reply]
Just got the same. There was a diff included in a commit message that made for a long commit message.
